internal/topology/variables/clusterclass_variable_validation_test.go

```diff
@@ -1023,12 +1023,266 @@ func Test_ValidateClusterClassVariable(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "Valid integer schema with CEL expression",
+			clusterClassVariable: &clusterv1.ClusterClassVariable{
+				Name: "cpu",
+				Schema: clusterv1.VariableSchema{
+					OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+						Type: "integer",
+						XValidations: []clusterv1.ValidationRule{{
+							Rule: "self >= 1",
+						}},
+					},
+				},
+			},
+		},
+		{
+			name: "Integer schema with invalid CEL expression",
+			clusterClassVariable: &clusterv1.ClusterClassVariable{
+				Name: "cpu",
+				Schema: clusterv1.VariableSchema{
+					OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+						Type: "integer",
+						XValidations: []clusterv1.ValidationRule{{
+							Rule: "this does not compile",
+						}},
+					},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "Forbid validation rules where cost total exceeds total limit",
```

Comment:
Are we hitting the per-expression limit here? As far as I can tell we don't check TotalCost at the moment

Reply:
Added more specific tests for the returned errors and looks like we're not :disappointed:. I'll look at that but now we have a test for it that makes it easier to know when it's been fixed.

Reply:
OK this is fixed now and it checks that the limit has been hit!